packages/access-token-jwt/test/dpop-verifier.test.ts

```diff
@@ -0,0 +1,1108 @@
+      url: 'https://api.example.com/resource',
+      headers: {
+        authorization: `DPoP ${jwt}`,
+        dpop: proof,
+      },
+      supportedAlgorithms: ['ES256'],
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Missing "jti" in DPoP proof');
+  });
+
+  it('throws InvalidProofError if "ath" claim is missing in DPoP proof', async () => {
+    const { publicKey, privateKey } = await generateKeyPair('ES256');
+
+    const jwt = 'header.payload.signature'; // access token placeholder
+
+    const proof = await new SignJWT({
+      htm: 'POST',
+      htu: 'https://api.example.com/resource',
+      iat: Math.floor(Date.now() / 1000),
+      jti: crypto.randomUUID(),
+      // ath is intentionally omitted
+    })
+      .setProtectedHeader({
+        alg: 'ES256',
+        typ: 'dpop+jwt',
+        jwk: await exportJWK(publicKey),
+      })
+      .sign(privateKey);
+
+    const result = verifyDPoP({
+      jwt,
+      method: 'POST',
+      url: 'https://api.example.com/resource',
+      headers: {
+        authorization: `DPoP ${jwt}`,
+        dpop: proof,
+      },
+      supportedAlgorithms: ['ES256'],
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Missing "ath" claim in DPoP proof');
+  });
+
+  it('throws InvalidProofError if "ath" claim is not a string', async () => {
+    const { publicKey, privateKey } = await generateKeyPair('ES256');
+
+    const jwt = 'header.payload.signature';
+
+    const proof = await new SignJWT({
+      htm: 'POST',
+      htu: 'https://api.example.com/resource',
+      iat: Math.floor(Date.now() / 1000),
+      jti: crypto.randomUUID(),
+      ath: 12345 as any,
+    })
+      .setProtectedHeader({
+        alg: 'ES256',
+        typ: 'dpop+jwt',
+        jwk: await exportJWK(publicKey),
+      })
+      .sign(privateKey);
+
+    const result = verifyDPoP({
+      jwt,
+      method: 'POST',
+      url: 'https://api.example.com/resource',
+      headers: {
+        authorization: `DPoP ${jwt}`,
+        dpop: proof,
+      },
+      supportedAlgorithms: ['ES256'],
+      accessTokenClaims: {
+        cnf: {
+          jkt: await calculateJwkThumbprint(await exportJWK(publicKey)),
+        },
+      },
+      iatOffset: 300,
+      iatLeeway: 30,
+    });
+
+    await expect(result).rejects.toThrow(InvalidProofError);
+    await expect(result).rejects.toThrow('Invalid "ath" claim');
+  });
+
+  it('throws InvalidProofError if "ath" claim does not match the hash of the access token', async () => {
```

Comment:
Isn't this testing the same thing as the test case called `throws if ath does not match`?